FHEM/00_SIGNALduino.pm

```diff
@@ -2904,7 +2906,12 @@ sub SIGNALduino_Parse_MN {
 
   my $hash = shift // return;   #return if no hash  is provided
   my $rmsg = shift // return;   #return if no rmsg is provided
- 
+
+  my $mark;
+  if (substr($rmsg,5,1) eq 'Y') { # WMBus
```

Comment:
Ich verstehe grob, warum das Y hier entfernt wird.
Was ich aber nicht verstanden habe ist, wieso die Daten mit Y und nicht MN im Modul ankommen.
Gerade auch, weil die FW ja noch angepasst werden muss.

Reply:
>Was ich aber nicht verstanden habe ist, wieso die Daten mit Y und nicht MN im Modul ankommen.
Gerade auch, weil die FW ja noch angepasst werden muss.

Das Ganze ist **damals erbaut wurden um vom CUL verarbeitet zu werden**.
Die Nachrichten auf dem CUL sehen im INTERNAL RAWMSG wie folgt aus
- `b494493447941295119075CDD780DFF5F350082C600000F0007C113FF8F08FF49450100FF2C741401001F31553201D6D4000200010022002B0058008500EF007C6E2F0087007D00BE00B2002F046D33111A324D1A96`
- `b3E44934460458348231AA31C7801FD086C81027C034955230082026C2424FFFF81037C034C41230082036CFFFF03E0CFFD17000000326CFFFF046D3A111A3202F685FDAC7E0F002129CB`
- `bY25442D2C078018631B168D20BF10C06920E4EF559D60F16381ED39B1B7B71560CC7608844E269BD9`

Die Kennung Y wurde damals zusätzlich vom Entwickler genutzt um einen bestimmten Typ zu unterscheiden.
https://svn.fhem.de/trac/browser/trunk/fhem/FHEM/36_WMBUS.pm#L61
Das Ganze hat was mit dem Protokollaufbau zu tun. Es gibt verschiedene `FRAME_TYPEN` aber auch Varianten des Aufbaus. Da nicht jede Variante gleich verarbeitet wird, so wurde vermutlich die Kennung hinterher geschaffen.
Ich vermute, das Anfangs nur das `b` verarbeitet wurde und dann kam ein anderere Typ hinzu aber mit anderem Aufbau. Somit wurde einfach zur Unterscheidung `Y` ergänzt.

Das Modul
https://svn.fhem.de/trac/browser/trunk/fhem/FHEM/36_WMBUS.pm#L24
verarbeitet zur Kennung (Match) nur das `b` in der Erkennung.

**Kurzum, wir sollten es ebenso aufgrund der Kompatiblität behandeln** und umsetzten. Somit kannst du die RAWMSG vom CUL und dem SIGNALduino als identisch ansehen. Eine Unterscheidung wäre da nicht optimal.

Hier siehst du, das die RAWMSG dann dem MN - Format entspricht.
```
CUL_868Mhz_RAWMSG | b3E44F536892656000108F5D97AAC0020256B755DD9228B433D8D5C82F0618CAD0643C2E540A722C2DC9B8A0138CE0041A2DCD11DE10F800C0001090086B41E0098DB14011E070416C580::-52.5

signalduino_DMSG | b3E44F536892656000108F5D97AA9002025FB7AD4DF8710D69A3867E687B0633B1515F7BA8A6BE0C93649EC5593F476E7DD3E82F0470F800C0001090086B41E0098CE14011E070416C50018
signalduino_Protocol_ID | 134
signalduino_RAWMSG | MN;D=3E44F536892656000108F5D97AA9002025FB7AD4DF8710D69A3867E687B0633B1515F7BA8A6BE0C93649EC5593F476E7DD3E82F0470F800C0001090086B41E0098CE14011E070416C50018;R=24;A=0;
```

... Viel Input aber das Thema WM-BUS ist auch sehr umfangreich. Zu vergleichen mit einem "normalen" FSK Protokoll wie Lacrosse geht leider nicht.

Reply:
Aktuell schlagen deine Tests fehl,
```
2024-04-04T15:57:53.3264336Z     # Failed test 'Verify return value'
2024-04-04T15:57:53.3265950Z     # at /home/runner/work/RFFHEM/RFFHEM/t/FHEM/00_SIGNALduino/01_SIGNALduino_Parse_MN.t line 170.
2024-04-04T15:57:53.3267446Z     # +-----+----+-------+
2024-04-04T15:57:53.3268466Z     # | GOT | OP | CHECK |
2024-04-04T15:57:53.3269437Z     # +-----+----+-------+
2024-04-04T15:57:53.3270230Z     # | 3   | eq | 1     |
2024-04-04T15:57:53.3271066Z     # +-----+----+-------+
2024-04-04T15:57:53.3291431Z 
2024-04-04T15:57:53.3292572Z # Failed test 'checking Good MN data, without RSSI, with set attribute rfmode=PCA301 on dummyDuino'
2024-04-04T15:57:53.3294938Z # at /home/runner/work/RFFHEM/RFFHEM/t/FHEM/00_SIGNALduino/01_SIGNALduino_Parse_MN.t line 174.
```
weil
```
2024-04-04T15:57:59.5088770Z 2024.04.04 15:57:53 4: dummyDuino: Parse_MN, Error! method ConvBresser_lightning, checksumCalc:0xA2B1 != checksum:0x899E
2024-04-04T15:57:59.5090348Z 2024.04.04 15:57:53 4: dummyDuino: Parse_MN, Found 2-FSK Protocol id 133 -> WMBus_S
2024-04-04T15:57:59.5092330Z 2024.04.04 15:57:53 5: dummyDuino: Parse_MN, Decoded matched MN Protocol id 133 dmsg=b0405019E8700AAAAAAAA0F13AA16ACC0540AAA49C814473A2774D208AC0B0167 RSSI = -71
2024-04-04T15:57:59.5094792Z 2024.04.04 15:57:53 5: dummyDuino: Dispatch, b0405019E8700AAAAAAAA0F13AA16ACC0540AAA49C814473A2774D208AC0B0167, test ungleich: disabled
2024-04-04T15:57:59.5096881Z 2024.04.04 15:57:53 5: dummyDuino: Dispatch, b0405019E8700AAAAAAAA0F13AA16ACC0540AAA49C814473A2774D208AC0B0167, -71 dB, dispatch
2024-04-04T15:57:59.5098614Z 2024.04.04 15:57:53 5: dummyDuino: dispatch b0405019E8700AAAAAAAA0F13AA16ACC0540AAA49C814473A2774D208AC0B0167
2024-04-04T15:57:59.5100160Z 2024.04.04 15:57:53 5: WMBUS raw msg b0405019E8700AAAAAAAA0F13AA16ACC0540AAA49C814473A2774D208AC0B0167
2024-04-04T15:57:59.5101599Z 2024.04.04 15:57:53 2: WMBUS Error during LinkLayer parse:CRC check failed on link layer
2024-04-04T15:57:59.5103167Z 2024.04.04 15:57:53 3: dummyDuino: Unknown code b0405019E8700AAAAAAAA0F13AA16ACC0540AAA49C814473A2774D208AC0B0167, help me!
2024-04-04T15:57:59.5104821Z 2024.04.04 15:57:53 4: dummyDuino: Parse_MN, Found 2-FSK Protocol id 134 -> WMBus_T
2024-04-04T15:57:59.5106843Z 2024.04.04 15:57:53 5: dummyDuino: Parse_MN, Decoded matched MN Protocol id 134 dmsg=b0405019E8700AAAAAAAA0F13AA16ACC0540AAA49C814473A2774D208AC0B0167 RSSI = -71
2024-04-04T15:57:59.5108946Z 2024.04.04 15:57:53 5: dummyDuino: Dispatch, b0405019E8700AAAAAAAA0F13AA16ACC0540AAA49C814473A2774D208AC0B0167, test gleich
2024-04-04T15:57:59.5110959Z 2024.04.04 15:57:53 4: dummyDuino: Dispatch, b0405019E8700AAAAAAAA0F13AA16ACC0540AAA49C814473A2774D208AC0B0167, Dropped due to short time or equal msg
2024-04-04T15:57:59.5112964Z not ok 8 - checking Good MN data, without RSSI, with set attribute rfmode=PCA301 on dummyDuino {
2024-04-04T15:57:59.5114589Z     1..2
2024-04-04T15:57:59.5115097Z     not ok 1 - Verify return value
2024-04-04T15:57:59.5115746Z     ok 2 - No Warnings in logfile
2024-04-04T15:57:59.5116267Z }
```
die dispatchte PCA301 Nachricht
`0405019E8700AAAAAAAA0F13AA16ACC0540AAA49C814473A2774D208AC0B0167` ebenso bei den Protokollen 133 und 134 "anschlägt".

Das kann in der Praxis nicht zum tragen kommen, weil mit den Registerwerten von PCA301 kannst du nie die Protokolle 133 und 134 durchtesten. Entweder deine Tests müssen hier angepasst werden oder es muss eine andere Lösung her.
Ich kann auch nie die Registerwerte bei eingestellten WMBus mit den PCA301 testen.

Habt Ihr da eine Idee @sidey79 oder @elektron-bbs ?

Reply:
MIST falsche Nachricht gelöscht :-(

Reply:
> Was ich aber nicht verstanden habe ist, wieso die Daten mit Y und nicht MN im Modul ankommen.
Gerade auch, weil die FW ja noch angepasst werden muss.

Ich schreibe nochmal "nur" kurz die Antwort.
Das mit dem Y wird aus der Historie und der Vielfahlt des WMBus Protokolles entstanden sein.
Wegen der Kompatiblität müsste es so beibehalten bleiben weil es so der CUL empfängt.

```
CUL_RAWMSG | bY25442D2C628346771B168D20BE124C0E20D48C96EA64C8FCCC14A07A3A8840A30DE4F084AB8D80F7
or
CUL_868Mhz_RAWMSG | b3E44F536892656000108F5D97AAC0020256B755DD9228B433D8D5C82F0618CAD0643C2E540A722C2DC9B8A0138CE0041A2DCD11DE10F800C0001090086B41E0098DB14011E070416C580
or
signalGateway_DMSG | b3E44F536892656000108F5D97AA9002025FB7AD4DF8710D69A3867E687B0633B1515F7BA8A6BE0C93649EC5593F476E7DD3E82F0470F800C0001090086B41E0098CE14011E070416C50018
signalGateway_Protocol_ID | 134
signalGateway_RAWMSG | MN;D=3E44F536892656000108F5D97AA9002025FB7AD4DF8710D69A3867E687B0633B1515F7BA8A6BE0C93649EC5593F476E7DD3E82F0470F800C0001090086B41E0098CE14011E070416C50018;R=24;A=0;
```

Reply:
Hmm, das leuchtet mir noch nicht ein.
Das logische Modul braucht die Daten wie sie auch das CUL Modul übergibt.

Aber das SIGNALduino Modul braucht sie doch nur so, wie es der SIGNALduino übergibt.

Funktioniert beim IT Modul doch auch genau so.

Reply:
Habe die RAW MSGs erst jetzt gesehen, aber wenn ich es richtig verstehe würde der SIGNALduino wie sonst auch

`MN;D=304497264202231800087A5E0020A5D9CF9E719E36DB255C06F2AEC0722FC3853A3031BD85EF085BDAD29194136A02DD7E00FD;R=253;A=0;
`
übergeben.
Das bedeutet aber auch, dass hier nicht Y vom uC an das Modul übergeben wird, sondern der Y vom Modul an das logische Modul.. Richtig? In "SIGNALduino_Parse_MN" würde somit nie das gesuchte Y in der Variable rmsg ankommen.

Reply:
Kannst du versuchen, das was dir nicht einleuchtet genauer beschreiben. Ich möchte ungern Missverständnisse aufkommen lassen und dir es bestmöglich erläutern.

Verstehe es richtig, das das dich "stört" ?
Die CUL Nachricht sieht so aus
`bY25442D2C628346771B168D20BE124C0E20D48C96EA64C8FCCC14A07A3A8840A30DE4F084AB8D80F7`
und es würde eine solche
RAWMSG | `MN;D=304497264202231800087A5E0020A5D9CF9E719E36DB255C06F2AEC0722FC3853A3031BD85EF085BDAD29194136A02DD7E00FD;R=253;A=0;`
Nachricht als RAW herauskommen beim SIGNALduino?
Die DMSG für das Modul sieht dann so
`bY304497264202231800087A5E0020A5D9CF9E719E36DB255C06F2AEC0722FC3853A3031BD85EF085BDAD29194136A02DD7E00FD`

Reply:
> `MN;D=304497264202231800087A5E0020A5D9CF9E719E36DB255C06F2AEC0722FC3853A3031BD85EF085BDAD29194136A02DD7E00FD;R=253;A=0; `
> übergeben.

JA genau! Das würdest du auf der telnet Konsole sehen.
Das `Y` würdest du in der Parse sehen, da es nur unter "bestimmten" Umständen ergänzt wird oder nicht.
https://github.com/HomeAutoUser/RFFHEM/blob/master_WMBUS/FHEM/00_SIGNALduino.pm#L2911

Es gibt so viele Nachrichten unterschiedlichen Types, das mit den Registerwerten nicht nur EIN Gerät empfangen werden kann. ;-)

Reply:
@HomeAutoUser  Ja vermutlich haben wir ein Missverständnis. Das möchte ich gerne beseitigen.

Also ich nehme folgendes an:

1. uC sendet `"MN;D=304497264202231800087A5E0020A5D9CF9E719E36DB255C06F2AEC0722FC3853A3031BD85EF085BDAD29194136A02DD7E00FD;R=253;A=0;"` an das 00_SIGNALduino Modul.
2. Die Parse funktion ruft die parseMN Funktion auf. 
Dort soll folgende Zeile aufgerufen werden
` if (substr($rmsg,5,1) eq 'Y') { # WMBus`

Da kann aber nie ein Y sein , denn Y ist kein Hex wert, wie er in D= vorkommen kann.

Dass man die dMSG für die Folgeverarbeitung anpasst, das verstehe ich, aber die rmsg nach einem nicht Hex Wert zu durchsuchen. Vermutlich würde ich es besser verstehen, wenn ich wüsste ob und wie dieses "Y" vom uC an das 00_SIGNALduino Modul übermittelt werden soll.  Das habe ich halt aus den Testdaten auch nicht entnehmen können und folglich wird diese Bedingung niemals wahr.


Im Codecov kann ich es leider nicht nachsehen, vermutlich liegt das daran, dass Du aus deinem Repo die Testergebnisse nicht in diese Repo übermitteln darfst.

Reply:
Ich versuche es mal bestmöglich wiederzugeben.

- mit den Einstellungen des Receivers (CC110x oder RFM69) wird eine Nachricht empfangen wo das SyncWord passt (es handelt sich aber nicht um die schon "richtige" Nachricht
- danach wird diese Nachricht decodiert bzw. bei manchen verschiedenen Typen in eine 3 auf 6 Decodierung gegeben
- bei erfolgreicher Übereinstimmung, wird diese weiter verarbeitet
- Prüfung, welcher Typ von Nachrichtenpaket es ist und ggf. ein Y ergänzen (frameTypeB) -> alle anderen ohne Y Ausgabe
- Ausgabe vom uC im `MN,D=Format`
- SignalduinoParse, entfernt ggf `Y` um durch die Prüfung HEX werte zu gelangen
`if ($rmsg !~ /^MN;D=[0-9A-F]+;(?:R=[0-9]+;)?(?:A=-?[0-9]{1,3};)?$/) { # AFC cc1101 0x32 (0xF2): FREQEST – Frequency Offset Estimate from Demodulator
`
- danach wird das Y wieder hinzugefügt
- zur Übergabe an den Dispatch DMSG wird das `b` anhand der Protokolldefinition ergänzt (`preamble => 'b', | clientmodule => 'WMBUS',`)
- Nachricht `b........` oder `bY.....` wird dann an das Modul 36_WMBUS.pm übergeben, welches zusätzlich auch das Modul WMBus.pm nutzt

Nun hoffe ich, es richtig zusammengefasst zu haben ;-) aber da wird mich @elektron-bbs schon korrigieren wenn es falsch war.

PS: Nach empfangener Nachricht, heißt es noch nicht, das du die Daten siehst. Bei vielen benötigt man einen sogenannten EncyptedKey. Nach Eingabe im angelegten Device in FHEM erhält man dann erst die verwertbaren Daten.

PS2: Beispiel für ein Device mit unverschlüsselten Daten
```
READINGS:
     2024-04-04 23:43:23   LQI             128
     2024-04-04 23:43:23   LastReceiveAge  120
     2024-04-04 23:43:23   RSSI            -70
     2024-04-04 23:43:23   batteryState    ok
     2024-04-04 23:43:23   decryption_ok   1
     2024-04-04 23:43:23   is_encrypted    0
     2024-04-04 23:43:23   state           no errors
```

PS3: Beispiel für ein Device mit verschlüsselten Daten
```
READINGS:
     2024-04-04 20:44:45   IODev           CUL_868Mhz
     2024-04-04 23:37:34   LQI             128
     2024-04-04 23:37:34   LastReceiveAge  899
     2024-04-04 23:37:34   RSSI            -79
     2024-04-04 23:37:34   state           encrypted message and no aeskey provided
```

Reply:
Können wir so einen Datensatz mit dem Y bitte in die Testdaten aufnehmen.

Meiner Meinung nach, funktioniert es so wie hier vorgeschlagen nicht.

In SIGNALduino_Parse  wird kein Y entfernt, daher wird SIGNALduino_Parse_MN den Datensatz nicht verarbeiten.

Reply:
Ich klinke mich mal mit ein.
Das 'Y' wird hier entfernt:

```
  my $mark;
  if (substr($rmsg,5,1) eq 'Y') { # WMBus
    $mark = substr($rmsg,5,1);
    $rmsg =~ s/Y//g;
  }
```

und nach diversen Prüfungen wieder eingefügt:

```
  if ($mark) { # WMBus
    $rawData  = $mark . $rawData;
  }
```

Reply:
Hier ist dann vermutlich in der kommentierten Zeile?

Naja wie schon mehrfach angemerkt, so weit dürfte man mit einem Y im D Teil nie kommen.

Reply:
Mit "hier" ist genau diese Zeile gemeint: `$rmsg =~ s/Y//g;`

Und wie erklärst du dir dann, das die empfangenen Daten mit dieser Änderung bereits erfolgreich vom WMBus-Modul dekodiert werden?

Reply:
Okay, ich habe mir den code noch mal angesehen, vorher existiert kein Prüfung auf hexwerte, da lag ich falsch.

Ich fasse mal zusammen.

1. Wir prüfen, ob an Position ein Y steht.
2. Wenn ja, entfernen wir alle Y aus der Nachricht, auch wenn sie nicht an Position 5 vorkommen
3. Dann durchlaufen wir die Prüfung, ob wir einen validen Datensatz haben.
4. Anschließend hängen wir an rmsg wieder ein Y an, wenn wir vorher auch eines gefunden haben.

Mir stellt sich die Frage, wollen wir wirklich mehr als ein Y entfernen? Wenn es am Ende nur darum geht, das Y in $rawData zu behalten, würde ich eine andere Vorgehensweise vorschlagen, aber den Punkt habe ich nicht verstanden, da ich noch keinen Datensatz mit Y gefunden habe :(

Reply:
@elektron-bbs ich habe mich doch nicht vertan.

Anbei ein List 
```
Internals:
   DEF        ITW 18230242 0 8
   DeviceMedium Heat Cost Allocator
   DeviceType 8
   FUUID      65dde351-f33f-33fd-8443-2c1014d0097d25e6
   IODev      [CUL_868Mhz](http://192.168.2.20:8083/fhem?detail=CUL_868Mhz)
   IdentNumber 18230242
   LASTInputDev [signalGateway](http://192.168.2.20:8083/fhem?detail=signalGateway)
   MSGCNT     228
   Manufacturer ITW
   MessageEncoding CUL
   NAME       [WMBUS_ITW_18230242_0_8](http://192.168.2.20:8083/fhem?detail=WMBUS_ITW_18230242_0_8)
   NR         808
   STATE      <div style="text-align:left;">2024-04-08 19:46:07 - encrypted message and no aeskey provided</div>
   TYPE       [WMBUS](http://192.168.2.20:8083/fhem?detail=WMBUS)
   Version    0
   addr       ITW_18230242_0_8
   eventCount 228
   model      ITW_8_0
   signalGateway_DMSG bY304497264202231800087ADF0020A5F1152567A5DD615D6A697D10BDC7605701099FCAB45D1F65B754386537074256870900FC
   signalGateway_FREQAFC 0
   signalGateway_MSGCNT 228
   signalGateway_Protocol_ID 134
   signalGateway_RAWMSG MN;D=304497264202231800087ADF0020A5F1152567A5DD615D6A697D10BDC7605701099FCAB45D1F65B754386537074256870900FC;R=252;A=0;
   signalGateway_RSSI -76
   signalGateway_TIME 2024-04-08 19:46:07
   READINGS:
     2024-04-08 19:46:07   LQI             9
     2024-04-08 19:46:07   LastReceiveAge  1499
     2024-04-08 19:46:07   RSSI            -74
     2024-04-08 19:46:07   state           encrypted message and no aeskey provided
Attributes:
   room       99_Unbekannt
```
von dem HeizKostenGateway im Haus.

Zusätzlich auch ein List von meinem Dads Wasserzähler aus der Nachbarschaft.
```
Internals:
   DEF        KAM 63188007 27 22
   DeviceMedium Cold water
   DeviceType 22
   FUUID      65dcb7ac-f33f-e2f7-2f34-b5b72c6c5b5ce336
   IODev      [CUL](http://192.168.1.6:8083/fhem?detail=CUL)
   IdentNumber 63188007
   LASTInputDev [sduino_868](http://192.168.1.6:8083/fhem?detail=sduino_868)
   MSGCNT     30
   Manufacturer KAM
   MessageEncoding [CUL](http://192.168.1.6:8083/fhem?detail=CUL)
   NAME       [WMBUS_KAM_63188007_27_22](http://192.168.1.6:8083/fhem?detail=WMBUS_KAM_63188007_27_22)
   NR         374
   STATE      <div style="text-align:left;">2024-04-08 20:05:52 - Payload CRC check failed on ELL, wrong AES key?</div>
   TYPE       WMBUS
   Version    27
   addr       KAM_63188007_27_22
   eventCount 30
   model      KAM_22_27
   sduino_868_DMSG bY2C442D2C078018631B168D2043033E6B20B03E2CF639AE070655E9A373BAA0A242AB94B8CD571A8871A38E1A618BDB
   sduino_868_FREQAFC -8
   sduino_868_MSGCNT 30
   sduino_868_Protocol_ID 999
   sduino_868_RAWMSG MN;D=2C442D2C078018631B168D2043033E6B20B03E2CF639AE070655E9A373BAA0A242AB94B8CD571A8871A38E1A618BDB;R=219;A=-5;
   sduino_868_RSSI -92.5
   sduino_868_TIME 2024-04-08 20:05:52
   READINGS:
     2024-04-08 20:05:52   LQI             97
     2024-04-08 20:05:52   LastReceiveAge  16
     2024-04-08 20:05:52   RSSI            -132.5
     2024-04-08 20:05:52   state           Payload CRC check failed on ELL, wrong AES key?
Attributes:
   room       Umgebung
```

Beide Devices sind Nachrichten ohne `MN;Y` aber dafür DMSG `bY`
Der Stand ist unser heutiger.

Dispatcht beide mal 
`MN;D=Y304497264202231800087A5E0020A5D9CF9E719E36DB255C06F2AEC0722FC3853A3031BD85EF085BDAD29194136A02DD7E00FD;R=253;A=0;` ihr werdet ein Device mit DMSG bY erhalten und in RAWMSG nur eine reine MN Nachricht ohne Y.

@elektron-bbs  Ich bin der Meinung, wir hatten drüber gesprochen dies auf Einheitlichkeit im Programm in FHEM wegzunehmen oder wie kannst du dir das erklären?

Reply:
Ähmm, ich weiß nicht so recht, worauf du hinaus willst.
Das Gateway liefert die Nachricht "MN;D=Y304497...", die RAWMSG wird wahrscheinlich in den Hash geschrieben, nachdem das Y entfernt wurde und die DMSG wird um die Preamble 'b' erweitert und dann dispatcht.

Reply:
Ich habe mir den Quellcode jetzt nochmal genauer angesehen. Wahrscheinlich muss das 'Y' auch wieder vor $rmsg eingefügt werden, damit RAWMSG korrekt in den Hash geschrieben wird.

Man könnte natürlich auch das Regex anpassen und die _limit_to_hex ebenfalls... aber @sidey79 hatte ja vielleicht eh schon eine bessere Idee?

Reply:
Naja, wenn das Y vom Uc kommt, dann hätte ich vorgeschlagen die Regex anzupassen.
Den Teil der der an _limit_to_hex hätte ich mit einem substr von dem ersten Zeichen (das Y) entfernt.

Dann bleibt das Y in der rmsg enthalten.

Wäre aber schön, wenn wir einen Testdatensatz (rmsg) mit dem Y aufnehmen.
Die fehlt da und machte mich halt unsicher ob das vom uC so kommt.

Reply:
Ich habe mal Euch etwas zusammengestellt @sidey79 & @elektron-bbs 
```
-MSG1-
uC_RAWMSG
            MN;D=Y304497264202231800087AE30020A5C247707714E518602D984FE2BF82C023288C39460740DC47132AAE42791B8F48F2EB00FD;R=253;A=0;
Internal_DMSG
                bY304497264202231800087AE30020A5C247707714E518602D984FE2BF82C023288C39460740DC47132AAE42791B8F48F2EB00FD
Internal_RAMSG
             MN;D=304497264202231800087AE30020A5C247707714E518602D984FE2BF82C023288C39460740DC47132AAE42791B8F48F2EB00FD;R=253;A=0

-MSG2-
uC_RAWMSG
            MN;D=Y304497264202231800087AE40020A5D75DA00DA8A5A3E0E7A39CEFE3FA26FDC78E6C874675EFA951F6031F8E211C9E0BA500F7;R=247;A=0;
HTML_RAWMSG (uninteressant für das SIGNALduino Projekt)
                 Y304497264202231800087AE40020A5D75DA00DA8A5A3E0E7A39CEFE3FA26FDC78E6C874675EFA951F6031F8E211C9E0BA5
Internal_DMSG
                bY304497264202231800087AE40020A5D75DA00DA8A5A3E0E7A39CEFE3FA26FDC78E6C874675EFA951F6031F8E211C9E0BA500F7
Internal_RAMSG
             MN;D=304497264202231800087AE40020A5D75DA00DA8A5A3E0E7A39CEFE3FA26FDC78E6C874675EFA951F6031F8E211C9E0BA500F7;R=247;A=0;
```

Wie verfahren wir weiter oder wie gedenkst du @sidey79 solche Nachrichten an das entsprechende Modul weiterzugeben?

Reply:
Was stört dich jetzt eigentlich an den Nachrichten?
Das einzige, was mir auffällt, ist das "Internal_RAMSG", das nicht der Nachricht vom µC entspricht.
Oder werden die Nachrichten etwa gar nicht an das WMBus-Modul weitergegeben?

Reply:
Ich bin immer noch dafür, dass eine rmsg Einzug in die Testdaten erhält.
Dann können wir die Implementierung so anpassen, dass die Tests erfolgreich sind.

Reply:
@sidey79 

>Ich bin immer noch dafür, dass eine rmsg Einzug in die Testdaten erhält.
Dann können wir die Implementierung so anpassen, dass die Tests erfolgreich sind.

https://github.com/RFD-FHEM/RFFHEM/pull/1248/files#diff-092416c5806fd44b768736cdcc4ed6b8b54b0c3f6123d65e96e3cc6f41b38c86R143-R174

@elektron-bbs 
> Oder werden die Nachrichten etwa gar nicht an das WMBus-Modul weitergegeben?
Doch diese werden an das Modul weitergegeben aber bei jeder _RMSG ist das Phänomen so.